### PR TITLE
feat: expose responseType

### PR DESCRIPTION
**Pull request by P0lip**
## Checklist

- [x] Tests have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## What kind of change does this PR introduce?

Feature

## What is the current behavior? What is the new behavior?

responseType is included

## Does this PR introduce a breaking change?

No

### CODE REVIEW

**Comment by philsturgeon** [comment]
Wait _should_ it be empty for all these instances? Should it not be `"json"`?

File: packages/http/src/forwarder/HttpForwarder.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export class HttpForwarder implements IForwarder<IHttpOperation, IHttpRequest, I
       statusCode: response.status,
       headers: response.headers,
       body: response.data,
+      responseType: (response.request && response.request.responseType) || '',
```
Comment (XVincentX) [inline packages/http/src/forwarder/HttpForwarder.ts]: Can really the request be undefined? This is really weird. I'd say it's a problem in the types.

File: packages/http/src/mocker/HttpMocker.ts [hunk 2]
```diff
@@ -72,6 +72,7 @@ export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpCo
         'Content-type': negotiationResult.mediaType,
       },
       body: mockedBody,
+      responseType: '', // todo(Vincenzo): to be implemented
```
Comment (XVincentX) [inline packages/http/src/mocker/HttpMocker.ts]: I have no idea what to put here in a mocked response to be honest.
The spec gives some value that I will probably never going to use; it does not even support XML. 

@philsturgeon

File: packages/http/src/forwarder/HttpForwarder.ts (hunk 1 above)
Comment (P0lip) [reply packages/http/src/forwarder/HttpForwarder.ts]: It's `any`, so well, might be a null.

**Pull request merged**